### ISSUE TITLE
My notes about possible improvements from Euroscipy tutorial

### ISSUE THREAD

**Issue by lesteve** [INRIA/scikit-learn-mooc#4]
Moved from: https://github.com/lesteve/scikit-learn-tutorial/issues/3

This is not very structured, so feel free to edit, comment, open other issues for bigger chunks of work:

### Content
- [ ] have a TOC per notebook?
- [ ] tinyurl (or huit.re) with link for easier access to the github repo (README)
- [ ] first notebook with TOC so that the binder goes directly to this notebook
- [x] Should we get rid of education-num everywhere, since this is the same as
  education?
- [x] Put solutions in different folder? They interfere with the notebooks, you have
  to say: open the 02 notebook but not the one with exercise ... 
- [x] naming: df vs adult_census harmonize, maybe data is good enough.
- [ ] Can we have link in notebooks to an other notebook, that works locally, on
  binder, in the MOOC platform, etc ...
- [x] Question about : pipeline with the scaler does it compute the mean on the
  training, so you have to explain how the Pipeline works, calls .fit and
  .transform. You don't have to explain maybe, you can just say the parameters
  are modified only in the .fit (so not in the .predict)
- [x] Question about pipeline, why is it useful rather than just writing the code
  yourself? You have to explain .fit and .fit_transform. Hmmm, maybe you can
  just add a comment about why the Pipeline is useful in general.
- [x] minor: sparse=False in OneHotEncoder just for visualization purposes (to see the numpy
  array).
- [ ] For exercise, have a link to the similar example, e.g. OrdinalEncoder put a
  link to what we did with OneHotEncoder.

- [x] Too-wide code: numerical_columns categorical_columns should cut at
  'capital-loss' and 'marital-status'. I think we should have a special formatter
  maybe black (I feel like it takes too much vertical space) or maybe yapf with
  some nice settings.
- [x] Say that education-num is not the number of years of education (I say that we
  could expect this, but I did not say this was not true)
- [x] young people work part-time. Say that non-working people (students) are not
  part of the survey.
- [x] Harmonize the way to get categorical_columns vs numerical_columns. Some code
  use dtype some code use explicit column names.
- [x] Side-comments about the train test split, goal is not to memorize. Should
  there be more details for the MOOC ? Or links to the first part about
  overfitting vs underfitting.
- [x] 02 exercise 01, not cross_val_score but use train_test_split
- [x] different kind of preprocessing, add a link to user guide. Question was: what
  happens if the data is not gaussian.
- [x] n_iter_ is a list for some reason ...
```
  print(
    f"The accuracy using a {model.__class__.__name__} is "
    f"{model.score(data_test, target_test):.3f} with a fitting time of "
    f"{elapsed_time:.3f} seconds in {model[-1].n_iter_} iterations"
  )
  The accuracy using a Pipeline is 0.818 with a fitting time of 0.809 seconds in [13] iterations
```
- [x] Cross-validation explanation plot: add legend for blue vs red. It looks like
  there might be better images from scikit-learn documentation.
- [x] handle_unknown='ignore': explain more the reason: to put 0 in the categories if at test time, a  category has not been seen in the train data.

### Miscellaneous
- [x] Timings are very slow on binder
  0.7 s for LogisticRegression fit vs 5.6s on binder.
  2 minutes (rather than ~10s on Olivier's machine) for Reference pipeline (no
  numerical scaling and integer-coded categories)
  02_basic_preprocessing_exercise_03_solution.ipynb

**Comment by lesteve** [comment]
Closing I feel most of it has been addressed and what remains is not that crucial or vague.
